### PR TITLE
Fixed bugs in memory profiler

### PR DESCRIPTION
**Pull request by n8o**
Fixed some scope issues in onRealloc{}. JS was throwing errors that it couldn't be found.
Added a check for isNaN so we don't throw off the calculations in the ui.
Fixed issues with the stack traces not rendering and throwing an error.
Fixed event listeners for ui. was using the wrong reference to this and not actually updating variables.

### CODE REVIEW

File: src/memoryprofiler.js [hunk 1]
```diff
@@ -117,7 +117,11 @@ var emscriptenMemoryProfiler = {
 
     // Decrement global stats.
     var sz = this.allocatedPtrSizes[ptr];
-    this.totalMemoryAllocated -= sz;
+    if(!isNaN(sz))
```
Comment (kripken) [inline src/memoryprofiler.js]: `{` should be on the same line. also space after `if`.

**Comment by kripken** [comment]
I had some style comments, but leaving to @juj who wrote this code.

File: src/memoryprofiler.js [hunk 2]
```diff
@@ -173,9 +177,15 @@ var emscriptenMemoryProfiler = {
...
+      div.innerHTML = "<div style='border: 2px solid black; padding: 2px;'><canvas style='border: 1px solid black; margin-left: auto; margin-right: auto; display: block;' id='memoryprofiler_canvas' width='100%' height='50'></canvas>trackedCallstackMinSizeBytes=<input id='memoryprofiler_min_tracked_alloc_size' type=number value="+this.trackedCallstackMinSizeBytes+"></input><br/><input id='memoryprofiler_enable_allocation_stats' type='checkbox'>Print allocation statistics by callstack to html log (warning: slow!)</input><input id='memoryprofiler_clear_alloc_stats' type='button' value='Clear alloc stats' ></input><div id='memoryprofiler'></div>";
       document.body.appendChild(div);
       memoryprofiler = document.getElementById('memoryprofiler');
+
+      self = this;
```
Comment (juj) [inline src/memoryprofiler.js]: I think here `var self = this;` would be more appropriate?

File: src/memoryprofiler.js [hunk 3]
```diff
@@ -354,7 +364,8 @@ var emscriptenMemoryProfiler = {
         if (numcalls >= this.allocateStatisticsMinReported) calls.push(i);
       }
 
-      calls.sort(function(a,b) { return this.allocationSiteStatistics[b] - this.allocationSiteStatistics[a]; });
+      self = this;
```
Comment (juj) [inline src/memoryprofiler.js]: Also `var self = this;` to not pollute global namespace.

**Comment by juj** [comment]
Superb work! Thanks for poking on this, indeed there were quite a few things broken like you mention, my test didn't stress those :/

Can you make sure to add your name to `AUTHORS`, check the style comments, and close and reopen the pull request to `incoming` branch instead. Thanks!

**Pull request closed without merging**